### PR TITLE
Fix Coverity 1498612: integer overflow

### PR DESCRIPTION
**Pull request by paulidale**
For the case where `current` is zero, the shift becomes 32 which overflows an integer.

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
Ping @openssl/otc & @openssl/committers for reviews

**Comment by t8m** [comment]
This is really stupid Coverity here. The condition of the while loop is `current & 0xFFFF` that means inside the loop current cannot be zero. I am really against "fixing" this by the cast.

**Comment by t8m** [comment]
Same OTC decision needed as in #17894

**Comment by paulidale** [comment]
Agreed, I missed that current couldn't be zero.  Perhaps an assert would be better.  Either assert(current != 0) or assert(pos < 32).
Plus a comment saying that failure isn't possible?

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
ping @openssl/committers for second review

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
Merged to all three branches, thanks for the reviews.

**Pull request closed without merging**
